src/screens/VisualTests/VisualTests.tsx

```diff
@@ -150,104 +93,229 @@ export const VisualTestsWithoutSelectedBuildId = ({
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [JSON.stringify(buildStatusUpdate), updateBuildStatus]);
 
-  const shouldSwitchToLastBuildOnBranch =
-    canSwitchToLastBuildOnBranch && lastBuildOnBranchCompletedStory;
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [lastBuildOnBranchIsSelectable, JSON.stringify(lastBuildOnBranch), updateBuildStatus]);
 
-  // Ensure we are holding the right story build
+  // Auto-select the last build on branch if it's selectable and ready
   useEffect(() => {
     setSelectedBuildInfo((oldSelectedBuildInfo) =>
       updateSelectedBuildInfo(oldSelectedBuildInfo, {
-        shouldSwitchToLastBuildOnBranch,
+        shouldSwitchToLastBuildOnBranch: lastBuildOnBranchIsSelectable && lastBuildOnBranchIsReady,
         lastBuildOnBranchId: lastBuildOnBranch?.id,
         storyId,
       })
     );
-  }, [shouldSwitchToLastBuildOnBranch, lastBuildOnBranch?.id, storyId, setSelectedBuildInfo]);
+  }, [
+    lastBuildOnBranchIsSelectable,
+    lastBuildOnBranchIsReady,
+    lastBuildOnBranch?.id,
+    setSelectedBuildInfo,
+    storyId,
+  ]);
 
   const switchToLastBuildOnBranch = useCallback(
     () =>
-      canSwitchToLastBuildOnBranch &&
+      lastBuildOnBranch?.id &&
+      lastBuildOnBranchIsSelectable &&
       setSelectedBuildInfo({ buildId: lastBuildOnBranch.id, storyId }),
-    [setSelectedBuildInfo, canSwitchToLastBuildOnBranch, lastBuildOnBranch?.id, storyId]
+    [setSelectedBuildInfo, lastBuildOnBranchIsSelectable, lastBuildOnBranch?.id, storyId]
   );
 
-  return !selectedBuildHasCorrectBranch || !selectedBuild || storyDataIsStale || queryError ? (
+  return !selectedBuild || !hasSelectedBuild || !hasData || queryError ? (
     <NoBuild
       {...{
         queryError,
-        hasData: !!data && !storyDataIsStale,
-        hasProject: !!data?.project,
-        hasSelectedBuild: !!selectedBuildHasCorrectBranch && !!selectedBuild,
+        hasData,
+        hasProject,
+        hasSelectedBuild,
         startDevBuild,
         localBuildProgress,
         branch: gitInfo.branch,
         setAccessToken,
       }}
     />
   ) : (
-    <BuildResults
-      {...{
-        branch: gitInfo.branch,
-        dismissBuildError,
-        localBuildProgress,
-        ...(lastBuildOnBranch && { lastBuildOnBranch }),
-        lastBuildOnBranchCompletedStory,
-        ...(canSwitchToLastBuildOnBranch && { switchToLastBuildOnBranch }),
-        startDevBuild,
-        userCanReview,
-        isReviewing,
-        onAccept,
-        onUnaccept,
-        ...(selectedBuildHasCorrectBranch && { selectedBuild }),
-        setAccessToken,
-        storyId,
-      }}
-    />
+    <ReviewTestProvider watchState={reviewState}>
+      <SelectedBuildProvider watchState={selectedBuild}>
+        <BuildResults
+          {...{
+            branch: gitInfo.branch,
+            dismissBuildError,
+            localBuildProgress,
+            ...(lastBuildOnBranch && { lastBuildOnBranch }),
```

Comment:
Isn't this available on the context now?

Reply:
`lastBuildOnBranch` isn't, only `selectedBuild` is right now. I'll fix that though.

Reply:
Refactored `SelectedBuildContext` => `BuildContext` to store `lastBuildOnBranch` on there as well, along with the rest of `useBuild`'s data.